src/liballoc/boxed.rs

```diff
@@ -101,6 +101,45 @@ impl<T> Box<T> {
     }
 }
 
+impl<T : ?Sized> Box<T> {
+    /// Constructs a box from the raw pointer.
+    ///
+    /// After this function call, pointer is owned by resulting box. In
+    /// particular, it means that `Box` destructor calls destructor of
+    /// `T` and releases memory.
+    ///
+    /// Function is unsafe, because improper use of this function may
+    /// lead to memory problems like double-free, for example if the
+    /// function is called twice on the same raw pointer.
```

Comment:
It may be worth mentioning that the only valid pointers to pass to this function are those created from `boxed::into_raw`. I don't think we'd want to bake in any assumptions about how `Box` is allocated and letting others think that it can be used to free any old `malloc` memory.